tests/native_api_tests/test_api_model.cpp

```diff
@@ -351,37 +375,197 @@ TEST_CASE("API Model") {
+        std::vector<double> const input_sym_load_p_specified{0.0};
+        std::vector<double> const input_sym_load_q_specified{500.0};
+        Buffer input_sym_load_buffer{PGM_def_input_sym_load, 1};
+        input_sym_load_buffer.set_nan();
+        input_sym_load_buffer.set_value(PGM_def_input_sym_load_id, input_sym_load_id.data(), -1);
+        input_sym_load_buffer.set_value(PGM_def_input_sym_load_node, input_sym_load_node.data(), -1);
+        input_sym_load_buffer.set_value(PGM_def_input_sym_load_status, input_sym_load_status.data(), -1);
+        input_sym_load_buffer.set_value(PGM_def_input_sym_load_type, input_sym_load_type.data(), -1);
+        input_sym_load_buffer.set_value(PGM_def_input_sym_load_p_specified, input_sym_load_p_specified.data(), -1);
+        input_sym_load_buffer.set_value(PGM_def_input_sym_load_q_specified, input_sym_load_q_specified.data(), -1);
+
+        // input dataset - row
+        DatasetConst input_dataset_row{"input", 0, 1};
+        input_dataset_row.add_buffer("node", 1, 1, nullptr, input_node_buffer);
+        input_dataset_row.add_buffer("source", 1, 1, nullptr, input_source_buffer);
+        input_dataset_row.add_buffer("sym_load", 1, 1, nullptr, input_sym_load_buffer);
+
+        // input dataset - col
+        DatasetConst input_dataset_col{"input", 0, 1};
+        input_dataset_col.add_buffer("node", 1, 1, nullptr, nullptr);
+        input_dataset_col.add_attribute_buffer("node", "id", input_node_id.data());
+        input_dataset_col.add_attribute_buffer("node", "u_rated", input_node_u_rated.data());
+
+        input_dataset_col.add_buffer("source", 1, 1, nullptr, nullptr);
+        input_dataset_col.add_attribute_buffer("source", "id", input_source_id.data());
+        input_dataset_col.add_attribute_buffer("source", "node", input_source_node.data());
+        input_dataset_col.add_attribute_buffer("source", "status", input_source_status.data());
+        input_dataset_col.add_attribute_buffer("source", "u_ref", input_source_u_ref.data());
+        input_dataset_col.add_attribute_buffer("source", "sk", input_source_sk.data());
+        input_dataset_col.add_attribute_buffer("source", "rx_ratio", input_source_rx_ratio.data());
+
+        input_dataset_col.add_buffer("sym_load", 1, 1, nullptr, nullptr);
+        input_dataset_col.add_attribute_buffer("sym_load", "id", input_sym_load_id.data());
+        input_dataset_col.add_attribute_buffer("sym_load", "node", input_sym_load_node.data());
+        input_dataset_col.add_attribute_buffer("sym_load", "status", input_sym_load_status.data());
+        input_dataset_col.add_attribute_buffer("sym_load", "type", input_sym_load_type.data());
+        input_dataset_col.add_attribute_buffer("sym_load", "p_specified", input_sym_load_p_specified.data());
+        input_dataset_col.add_attribute_buffer("sym_load", "q_specified", input_sym_load_q_specified.data());
+
+        // update dataset
+        std::vector<Idx> input_source_indptr{0, 1, 2};
```

Comment:
Nitpick: Naming, isn't this `update_source_indptr`? Or am I wrong?

Same with `input_sym_load_indptr`.